src/lib.rs

```diff
@@ -237,33 +238,57 @@ mod tests {
 
     #[tokio::test]
     async fn test_encrypt_decrypt_128_gcm() {
-        let provider: SimpleKeyProvider<Aes128Gcm> = SimpleKeyProvider::init([1; 16]);
-        let cipher: EnvelopeCipher<Aes128Gcm, _> = EnvelopeCipher::init(provider);
+        let provider: SimpleKeyProvider = SimpleKeyProvider::init([1; 16]);
+        let cipher: EnvelopeCipher<_> = EnvelopeCipher::init(provider);
         test_encrypt_decrypt(cipher).await;
 
         let provider: SimpleKeyProvider<Aes128Gcm> = SimpleKeyProvider::init([1; 16]);
-        let provider: Box<dyn KeyProvider<_>> = Box::new(provider);
-        let cipher: EnvelopeCipher<Aes128Gcm, _> = EnvelopeCipher::init(provider);
+        let provider: Box<dyn KeyProvider<Cipher = _>> = Box::new(provider);
+        let cipher: EnvelopeCipher<_> = EnvelopeCipher::init(provider);
         test_encrypt_decrypt(cipher).await;
     }
 
     #[tokio::test]
     async fn test_encrypt_decrypt_256_gcm() {
         let provider: SimpleKeyProvider<Aes256Gcm> = SimpleKeyProvider::init([1; 16]);
-        let cipher: EnvelopeCipher<Aes256Gcm, _> = EnvelopeCipher::init(provider);
+        let cipher: EnvelopeCipher<_> = EnvelopeCipher::init(provider);
         test_encrypt_decrypt(cipher).await;
 
         let provider: SimpleKeyProvider<Aes256Gcm> = SimpleKeyProvider::init([1; 16]);
-        let provider: Box<dyn KeyProvider<_>> = Box::new(provider);
-        let cipher: EnvelopeCipher<Aes256Gcm, _> = EnvelopeCipher::init(provider);
+        let provider: Box<dyn KeyProvider<Cipher = _>> = Box::new(provider);
+        let cipher: EnvelopeCipher<_, _> = EnvelopeCipher::init(provider);
+        test_encrypt_decrypt(cipher).await;
+    }
+
+    #[tokio::test]
+    async fn test_encrypt_decrypt_128_gcm_siv() {
+        let provider: SimpleKeyProvider<Aes128GcmSiv> = SimpleKeyProvider::init([1; 16]);
+        let cipher: EnvelopeCipher<_, _> = EnvelopeCipher::init(provider);
+        test_encrypt_decrypt(cipher).await;
+
+        let provider: SimpleKeyProvider<Aes128GcmSiv> = SimpleKeyProvider::init([1; 16]);
+        let provider: Box<dyn KeyProvider<Cipher = _>> = Box::new(provider);
+        let cipher: EnvelopeCipher<_, _> = EnvelopeCipher::init(provider);
+        test_encrypt_decrypt(cipher).await;
+    }
+
+    #[tokio::test]
+    async fn test_encrypt_decrypt_256_gcm_siv() {
+        let provider: SimpleKeyProvider<Aes256GcmSiv> = SimpleKeyProvider::init([1; 16]);
+        let cipher: EnvelopeCipher<_, _> = EnvelopeCipher::init(provider);
+        test_encrypt_decrypt(cipher).await;
+
+        let provider: SimpleKeyProvider<Aes256GcmSiv> = SimpleKeyProvider::init([1; 16]);
+        let provider: Box<dyn KeyProvider<Cipher = _>> = Box::new(provider);
+        let cipher: EnvelopeCipher<_, _> = EnvelopeCipher::init(provider);
```

Comment:
Should this just take 1 generic param now?

Reply:
Yes it should. Not mandatory, but it should. I'll fix this in separate PR.